File: java/src/test/java/problems1/IntegerToRomanTest.java
package problems1;

import org.junit.Test;
import testing.Sample;
import testing.SampledTest;

import static com.google.common.truth.Truth.assertWithMessage;

public class IntegerToRomanTest extends SampledTest {
    @Override
    public String getFileName() {
        return "integer_to_roman.json";
    }

    @Test
    public void test() {
        for (IntegerToRoman solution : IntegerToRoman.values()) {
            for (TypedSample sample : getSamples(TypedSample[].class)) {
                assertWithMessage(sample.getMessage())
                    .that(solution.intToRoman(sample.input.num))
                    .isEqualTo(sample.output);
            }
        }
    }

    public static class TypedSample extends Sample<TypedSample.Input, String> {
        public static class Input {
            public int num;
        }
    }
}

File: javascript/__tests__/problems1_100/integer-to-roman.test.js
import {strictEqual} from 'assert';
import SOLUTIONS from '../../src/problems1_100/integer-to-roman';
import getSamples from '../../../testing/js/src/sample/samples';

describe(
    'IntegerToRoman',
    () => {
      test(
          'test',
          () =>
              SOLUTIONS.forEach(solution => {
                getSamples().forEach(sample => {
                  strictEqual(
                      solution.intToRoman(sample.input.num),
                      sample.output,
                      sample.message,
                  );
                });
              }),
      );
    },
);
